### ISSUE TITLE
OutOfMemoryException due to an invalid IL produced for accessing to the field in a generic class

### ISSUE THREAD

**Issue by rameel** [ltrzesniewski/InlineIL.Fody#5]
Accessing to the member in a generic class produces an invalid IL that makes the Runtime throws an OutOfMemoryException

```cs
public class Program
{
    public static void Main() => Console.WriteLine(Sample<int>.GetValue());
}
public static class Sample<T>
{
    public static readonly int Value = 10;

    public static int GetValue()
    {
        Ldsfld(new FieldRef(typeof(Sample<T>), nameof(Value)));
        return IL.Return<int>();
    }
}
```

.NET Core 2.2.1  (SDK 2.2.102)

**Comment by ltrzesniewski** [comment]
Thanks for the report! It looks like the generic parameter is not propagated properly...
I added a failing test in 619028e6ae261014e76cea9ac6f65bc7d9e88e98 for now, I'll fix the issue later.

**Comment by ltrzesniewski** [comment]
Fixed in v1.0.3.

**Comment by rameel** [comment]
Thanks!
